src/v2/stack_aggregator.py

```diff
@@ -361,3 +369,181 @@ def execute(request: Dict, persist=True):
+
+    def _get_package_details_with_vulnerabilities(self) -> List[Dict[str, object]]:
+        """Get package data from graph along with vulnerability."""
+        time_start = time.time()
+        pkgs_with_vuln = {
+            "result": {
+                "data": []
+            }
+        }
+        query = """
+                epv = [];
+                packages.each {
+                    g.V().has('pecosystem', ecosystem).
+                    has('pname', it.name).
+                    has('version', it.version).as('version', 'vuln').
+                    select('version').in('has_version').dedup().as('package').
+                    select('package', 'version', 'vuln').
+                    by(valueMap()).
+                    by(valueMap()).
+                    by(out('has_snyk_cve').valueMap().fold()).
+                    fill(epv);
+                }
+                epv;
+                """
+        # get rid of leading white spaces
+        query = inspect.cleandoc(query)
+        bindings = {
+            'ecosystem': self._normalized_packages.ecosystem,
+            'packages': []
+        }
+        # call gremlin in batches of GREMLIN_QUERY_SIZE
+        for pkgs in _get_packages_in_batch(self._normalized_packages.all_deps_without_pseudo,
+                                           GREMLIN_QUERY_SIZE):
+            # convert Tuple[Package] into List[{name:.., version:..}]
+            bindings['packages'] = [pkg.dict(exclude={'dependencies'}) for pkg in pkgs]
+
+            started_at = time.time()
+
+            result = post_gremlin(query, bindings)
+
+            logger.info(
+                '%s took %0.2f secs for post_gremlin() batch request',
+                self._request.external_request_id, time.time() - started_at)
+            if result:
+                pkgs_with_vuln['result']['data'] += result['result']['data']
+
+        logger.info('%s took %0.2f secs for get_package_details_with_'
+                    'vulnerabilities() for total_results %d', self._request.external_request_id,
+                    time.time() - time_start, len(pkgs_with_vuln['result']['data']))
+        return pkgs_with_vuln['result']['data']
+
+    def get_package_details_from_graph(self) -> Dict[Package, PackageDetails]:
+        """Get dependency data from graph."""
+        graph_response = self._get_package_details_with_vulnerabilities()
+        package_details: List[Tuple[Package, PackageDetails]] = []
+        for pkg in graph_response:
+            package_details.append(self._get_package_details(pkg))
+
+        psedo_pkgs_data = self._get_batch_sa_data_for_pseudo_version()
+        for pseudo_pkg in psedo_pkgs_data:
+            pseudo_pkg_details = self._get_golang_package_details(pseudo_pkg)
+            package_details.append(pseudo_pkg_details)
+
+        # covert list of (pkg, package_details) into map
+        return dict(package_details)
+
+    def _get_data_from_db(self, packages, query, caller=None) -> Dict:
```

Comment:
nit: Just to keep consistency in naming.
```suggestion
    def _get_data_from_graph(self, packages, query, caller=None) -> Dict:
```

Reply:
Ack